internal/core/block/block.go

```diff
@@ -103,6 +103,8 @@ type Block struct {
 	Delta crdt.CRDT
 	// Links are the links to other blocks in the DAG.
 	Links []DAGLink
+	// IsEncrypted is a flag that indicates if the block's delta is encrypted.
+	IsEncrypted *bool
```

Comment:
todo: Please don't use a pointer for a boolean, it feels very wasteful and confusing - suggesting that it's pointerness may be used for odd mutation reasons.  If you want it to be optional, please use `immutable.Option` instead, it is much cheaper, less confusing, and wont contribute to GC pressure.

If you have a very specific reason for using a pointer here, please document it.

Reply:
documented